pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -199,6 +200,15 @@ func (c *Cluster) checkHealth() error {
 			continue
 		}
 
+		//retry and don't failover if mon POD is not scheduled.
+		isScheduled, err := isMonScheduled(c.context.Clientset, c.Namespace, mon.Name)
+		if err != nil {
+			logger.Warningf("failed to check if mon %q is assigned to a node. %v", mon.Name, err)
+		} else if !isScheduled {
+			logger.Warningf("mon %q NOT found in quorum after timeout. Not failing over as the mon is not scheduled to any node", mon.Name)
+			continue Loop
```

Comment:
Instead of continuing the loop, we would need to reset the `c.monTimeoutList[mon.Name]` timestamp, then `return nil`. A timer triggers a new health check every 45s (by default) as seen up on line 85.

Reply:
Thanks for that suggestion.  Now resetting the `c.monTimeoutList[mon.Name]` map.